src/persist-client/src/impl/machine.rs

```diff
@@ -316,6 +316,17 @@ where
                         new_state
                     );
                     self.state = new_state;
+
+                    // Bound the number of entries in consensus.
+                    //
+                    // It's weird that this will return an error for the whole
+                    // request after we know it's successful, but this goes away
+                    // in #12223.
+                    let () = self
```

Comment:
What is this let binding trying to express? The way it is written it's as if it destructures the unit type

Reply:
it prevents things from compiling if we haven't unwrapped all the results. there's the unused Result warning, but I like that this is a bit more explicit